### ISSUE TITLE
check-stack-names pre-commit hook prints help message multiple times

### ISSUE THREAD

**Issue by zaro0508** [Sage-Bionetworks-IT/pre-commit-provisioner#9]
When running `check-stack-names` as a pre-commit hook the info message `A stack name can contain only...` is printed for ever invalid stack name.  It should only be printed once at the end.  This only happens when running as a pre-commit hook.  When running the script stand alone the message only appears once at the end.

```
➜ pre-commit run --all-files
Stack name linter........................................................Failed
- hook id: check-stack-names
- exit code: 1

- 'cvat_ubuntu' is an invalid stack name (in 'config/prod/cvat-ubuntu.yaml')
A stack name can contain only alphanumeric characters (case-sensitive) and hyphens. It must start with an alphabetic character and can't be longer than 128 characters.
https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/cfn-using-console-create-stack-parameters.html

- 'zaro_irc' is an invalid stack name (in 'config/prod/zaroirc.yaml')
A stack name can contain only alphanumeric characters (case-sensitive) and hyphens. It must start with an alphabetic character and can't be longer than 128 characters.
https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/cfn-using-console-create-stack-parameters.html
```

**Comment by cascadianblue** [comment]
If this is happening for the reason I think it is, then I'm not sure how to fix it.

As I understand, when pre-commit is passed more than a certain number of files (e.g. when you call `pre-commit run --all-files`), pre-commit will split the files into chunks and call the hook one time for each chunk.

Imagine you are committing 7 files, `a.yaml`, `b.yaml`, `c.yaml`, `d.yaml`, `e.yaml`, `f.yaml`, & `g.yaml`. Supposing pre-commits chunk size is 5, pre-commit will run `HOOK-NAME a.yaml b.yaml c.yaml d.yaml e.yaml` and then run `HOOK-NAME e.yaml f.yaml`.

I don't think this behavior is formally documented, but I found some discussion about it here:
https://github.com/pre-commit/pre-commit/issues/691#issuecomment-360224845

**Comment by cascadianblue** [comment]
I suppose we could use some kind of inter-process communication to make sure that the message only gets displayed once, but I think the effort and extra maintainability cost would not be worth it.

As long as the user is only committing a handful of files at a time (with the hooks installed, as opposed to running `pre-commit run --all-files`), this issue should never even come up.

**Comment by zaro0508** [comment]
how about we just remove the duplication message?
```A stack name can contain only alphanumeric characters (case-sensitive) and hyphens. It must start with an alphabetic character and can't be longer than 128 characters.
https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/cfn-using-console-create-stack-parameters.html```

**Comment by zaro0508** [comment]
this is fixed with PR #10
